src/DurableTask.Netherite/TransportLayer/EventHubs/EventHubsProcessor.cs

```diff
@@ -433,7 +433,11 @@ async Task IEventProcessor.ProcessEventsAsync(PartitionContext context, IEnumera
                 {
                     this.traceHelper.LogDebug("EventHubsProcessor {eventHubName}/{eventHubPartition}({incarnation}) is processing packets", this.eventHubName, this.eventHubPartition, current.Incarnation);
 
-                    await foreach ((EventData eventData, PartitionEvent[] events, long seqNo) in this.blobBatchReceiver.ReceiveEventsAsync(this.taskHubGuid, packets, current.ErrorHandler.Token, current.NextPacketToReceive.seqNo))
+                    // we need to update the next expected seqno even if the iterator returns nothing, since it may have discarded some packets.
+                    // iterators do not support ref arguments, so we use a simple wrapper class to work around this limitation
+                    MutableLong nextPacketToReceive = new MutableLong() { Value = current.NextPacketToReceive.seqNo };
+
+                    await foreach ((EventData eventData, PartitionEvent[] events, long seqNo) in this.blobBatchReceiver.ReceiveEventsAsync(this.taskHubGuid, packets, current.ErrorHandler.Token, nextPacketToReceive))
```

Comment:
We also call `ReceiveEventsAsync` in `LoadMonitorProcessor` and `EventHubsTransport`. What is the reason why we don't need to use update the expected `seqno` in there?

Reply:
Those two are used by clients and the load monitor, respectively. The latter do not care about the order or about duplication of received events, so they are not tracking the sequence numbers.